src/main/java/frc/robot/subsystems/DeployingBar.java

```diff
@@ -0,0 +1,71 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.DeployingBarConstants;
+import frc.robot.util.MotorFactory;
+
+public class DeployingBar extends SubsystemBase {
+
+  private final TalonFX m_motor1;
+  private final TalonFX m_motor2;
+
+  private double setpoint;
+  private PIDController m_pid;
+  private boolean isEnabled;
+
+  public DeployingBar() {
+    m_motor1 = MotorFactory.createTalonFX(DeployingBarConstants.kLeftMotor, getName());
+    m_motor2 = MotorFactory.createTalonFX(DeployingBarConstants.kRightMotor, getName());
+    m_pid = new PIDController(DeployingBarConstants.kP, DeployingBarConstants.kI, DeployingBarConstants.kD);
```

Comment:
For this subsystem, I thought there would be limit switches (or hall effect) that just detect if the bar is extended. Then we just power the bar until it hits the limit switch. A PID would work but would require more tuning.

Reply:
To be honest, Allan is MIA every class and they are completely redesigning the subsystem so I am just doing PID for now

Reply:
Kaushik is suggesting using motor current to detect this, what are your thoughts about that?

Reply:
Kind of a bruh moment wrt the extending bar CAD, I guess for now we can assume a PID, because I realized depending on how they make it, it might be hard to put a limit switch. 
I'm not sure motor current would work well, current can be noisy, current can also change if there are mechanical issues, and it would mean it would mess up if it felt any pressure (ex. hitting a wall, getting hit by another robot). Maybe we can test it, but I don't think it's a good idea.